lib/chef/resource/cron/cron_d.rb

```diff
@@ -115,14 +115,25 @@ def after_created
         raise ArgumentError, "The 'cookbook' property for the cron_d resource is no longer supported now that it ships as a core resource." if cookbook
       end
 
+      def validate_name
+        # ensure valid cron job names for linux, otherwise the jobs won't be executed
+        if linux?
+          unless cron_name =~ /^[a-zA-Z0-9_-]+$/
+            raise ArgumentError, "The cron job name should contain letters, numbers, hyphens and underscores only."
+          end
+        end
+      end
+
```

Comment:
why isn't this in `validate_resource_requirements` ? That is exactly what that is for, no?

Reply:
Thanks for the suggestion. I was not aware of `define_resource_requirements` and fixed it.